create 2-3 new Documents for VFS,VGZ,SEA documents (#21)
CCSMB/Standards: Standards/CCSMB-X.md

# CCSMB-X: Virtual File System Archive

*Author: walksanator (Walker Fowlkes)*

*Version: 1.0.*
     
*Last revised: 2023-01-DD*

## Quick information

| Information |                              |
| ----------- | -----------------------------|
| Version     | 1.0.9                        |
| Author      | walksanator (Walker Fowlkes) |

> | Type        | Virtual File System Archiv   |

Reviewer: Typo.
```suggestion
| Type        | Virtual File System Archive   |
```

| MIME        | `appilcation/vfs`            |
| Extensions  | `.vfs`                       |

## Technical details

> The VFS file format is a text-based format that represents folder(s) as a lua table

Reviewer 2: ```suggestion
The VFS file format is a text-based format that represents folder(s) as a Lua table
```

> where the keys of the table are used as file names and values are used for conents.

Reviewer 2: ```suggestion
where the keys of the table are used as file names and values are used for contents.
```

> Directories in VFS are tables, whereas files have their contents embedded as a string.

Reviewer 2: ```suggestion
Directories in VFS are tables, whereas files have their contents embedded as strings.
```

Author: fixed

> VFS files should be read and written in binary mode as to prevent converting binary charachters to `?`

Reviewer 2: ```suggestion
VFS files should be read and written in binary mode as to prevent converting binary characters to `?`.
```

> One drawback of the VFS format is that it does not save any MetaData of files compressed

Reviewer 2: I'd reword this to be a bit more positive:

> Note that the VFS format does not store metadata, so applications requiring metadata may need to use an alternative.

> ## Examples included

Reviewer 2: I'd recommend having the examples embedded inline with descriptions of what they do. This makes it easier to read quickly.

Author: but it is a full program, or should i have the Examples section contains example functions for creating/extracting (the gen_disk and unpack_vfs functions) along wit a small text example
and have the full example program and example image in the folder

Reviewer 2: You shouldn't be posting the full program here - put that on your own repo and link to it as an additional example. The examples here should be limited to single functions that demonstrate just one part at a time.

Author: so should i embed the example functions i use to create/unpack the archive with comments
but *just* those functions
and include a simpler example since having a folder and 1 lua script and a 18kb text file is a bit mutch

Reviewer 2: Sure, but try to make them as simple and easy to read as possible (e.g. assume the file's correct, no error checking).

Author: thats what they do allready
i do all the error checking outside of the functions anyway

Author: i think it is better now
